examples/Python/PET/asarray.py

```diff
@@ -0,0 +1,145 @@
+'''ImageData.asarray usage demo.
+
+Usage:
+Usage:
+  asarray.py [--help | options]
+
+Options:
+  -f <file>, --file=<file>     raw data file [default: my_forward_projection.hs]
+  -p <path>, --path=<path>     path to data files, defaults to data/examples/PET
+                               subfolder of SIRF root folder
+  -e <engn>, --engine=<engn>   reconstruction engine [default: STIR]
+  -s <stsc>, --storage=<stsc>  acquisition data storage scheme [default: memory]
+  --non-interactive            do not show plots
+'''
+
+## SyneRBI Synergistic Image Reconstruction Framework (SIRF)
+## Copyright 2015 - 2020 Rutherford Appleton Laboratory STFC
+## Copyright 2015 - 2017 University College London.
+##
+## This is software developed for the Collaborative Computational
+## Project in Synergistic Reconstruction for Biomedical Imaging (formerly CCP PETMR)
+## (http://www.ccpsynerbi.ac.uk/).
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+##   you may not use this file except in compliance with the License.
+##   You may obtain a copy of the License at
+##       http://www.apache.org/licenses/LICENSE-2.0
+##   Unless required by applicable law or agreed to in writing, software
+##   distributed under the License is distributed on an "AS IS" BASIS,
+##   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+##   See the License for the specific language governing permissions and
+##   limitations under the License.
+
+__version__ = '0.1.0'
+from docopt import docopt
+args = docopt(__doc__, version=__version__)
+
+#import math
+import numpy
+
+from sirf.Utilities import error, examples_data_path, existing_filepath
+
+# import engine module
+import importlib
+engine = args['--engine']
+pet = importlib.import_module('sirf.' + engine)
+
+# process command-line options
+data_file = args['--file']
+data_path = args['--path']
+if data_path is None:
+    data_path = examples_data_path('PET')
+storage = args['--storage']
+show_plot = not args['--non-interactive']
+
+# select acquisition data storage scheme
+# storage = 'file' (default):
+#     all acquisition data generated by the script is kept in
+#     scratch files deleted after the script terminates
+# storage = 'memory':
+#     all acquisition data generated by the script is kept in RAM
+#     (avoid if data is very large)
+scheme = pet.AcquisitionData.get_storage_scheme()
+if scheme != storage:
+    print('default storage scheme is %s' % repr(scheme))
+    print('setting storage scheme to %s' % repr(storage))
+    pet.AcquisitionData.set_storage_scheme(storage)
+else:
+    print('using default storage scheme %s' % repr(scheme))
+
+
+def test_img_asarray(img_data):
+    try:
+        img_asarray = img_data.asarray() # view
+        print('img_data.asarray() ok')
+    except Exception:
+        print('data not contiguous, making a copy...')
+        img_data = img_data.copy()
+        img_asarray = img_data.asarray()
```

Comment:
yeah well, except that the tests below are then really trivially satisfied.

In any case,  I think we need some tests that this is really a view (i.e. modify `img_asarray`, check if it changed `img_data`. But those tests should not be in an example.

Reply:
I think we need a unit test for this sort of thing:

```py
img_data.asarray()[0,0,0] = random()
assert img_data[0,0,0] == img_data.asarray()[0,0,0]
```